cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -97,7 +97,13 @@ protected void checkSystemServices() throws ServiceUnavailableException {
     Callable<Boolean> monitorCallable = new Callable<Boolean>() {
       @Override
       public Boolean call() throws Exception {
-        return getMonitorClient().allSystemServicesOk();
+        // first wait for all system services to be 'OK'
+        if (!getMonitorClient().allSystemServicesOk()) {
+          return false;
+        }
+        // check that the dataset service is up
+        getNamespaceClient().list();
```

Comment:
the list() method throws an IOE, which will be retried anyways.
The http client for the namespace client will log all attempted requests and their response, which I think is enough logging. There's no need to additionally log (such as you've done in your comment).